### DISCUSSION TITLE
Build for Embedded STLinux (32 bit)

### DISCUSSION THREAD

**Discussion by symdeb** [duckdb/duckdb#14186] [Ideas]
STMicro provides a develop package with build environment for STM32MP1. While on Ubutu 24.04 building from source is successful,  it fails when building duckdb cross compile from source for  ARM.  Request to add a feature to build for ARM 32bit or guidance on build process.

$CC -v
Using built-in specs.
COLLECT_GCC=**arm-ostl-linux-gnueabi-gcc**
COLLECT_LTO_WRAPPER=/media/user/H3D2P1/Develop/STMICRO/STM_ECO_5.0.0/Developer-Package/SDK/sysroots/x86_64-ostl_sdk-linux/usr/libexec/arm-ostl-linux-gnueabi/gcc/arm-ostl-linux-gnueabi/12.2.0/lto-wrapper
Target: arm-ostl-linux-gnueabi
Configured with: ../../../../../../work-shared/gcc-12.2.0-r0/gcc-12.2.0/configure --build=x86_64-linux --host=x86_64-ostl_sdk-linux --target=arm-ostl-linux-gnueabi --prefix=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr --exec_prefix=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr --bindir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr/bin/arm-ostl-linux-gnueabi --sbindir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr/bin/arm-ostl-linux-gnueabi --libexecdir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr/libexec/arm-ostl-linux-gnueabi --datadir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr/share --sysconfdir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/etc --sharedstatedir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/com --localstatedir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/var --libdir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr/lib/arm-ostl-linux-gnueabi --includedir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr/include --oldincludedir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr/include --infodir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr/share/info --mandir=/usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-ostl_sdk-linux/usr/share/man --disable-silent-rules --disable-dependency-tracking --with-libtool-sysroot=/local/mpuostlqa/gnbsx51335/workspace/ostl-build-cache/workdir/ostl/build/tmp-glibc/work/x86_64-nativesdk-ostl_sdk-linux/gcc-cross-canadian-arm/12.2.0-r0/recipe-sysroot --with-gnu-ld --enable-shared --enable-languages=c,c++ --enable-threads=posix --enable-multilib --enable-c99 --enable-long-long --enable-symvers=gnu --enable-libstdcxx-pch --program-prefix=arm-ostl-linux-gnueabi- --without-local-prefix --disable-install-libiberty --disable-libssp --enable-libitm --enable-lto --disable-bootstrap --with-system-zlib --with-linker-hash-style=gnu --enable-linker-build-id --with-ppl=no --with-cloog=no --enable-checking=release --enable-cheaders=c_global --without-isl --with-gxx-include-dir=/not/exist/usr/include/c++/12.2.0 --with-build-time-tools=/local/mpuostlqa/gnbsx51335/workspace/ostl-build-cache/workdir/ostl/build/tmp-glibc/work/x86_64-nativesdk-ostl_sdk-linux/gcc-cross-canadian-arm/12.2.0-r0/recipe-sysroot-native/usr/arm-ostl-linux-gnueabi/bin --with-sysroot=/not/exist --with-build-sysroot=/local/mpuostlqa/gnbsx51335/workspace/ostl-build-cache/workdir/ostl/build/tmp-glibc/work/x86_64-nativesdk-ostl_sdk-linux/gcc-cross-canadian-arm/12.2.0-r0/recipe-sysroot --enable-standard-branch-protection --with-plugin-ld=ld --enable-poison-system-directories --disable-static --enable-nls --with-glibc-version=2.28 --enable-initfini-array
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 12.2.0 (GCC) 

EN=ninja make
mkdir -p ./build/release && \
cd build/release && \
cmake          -DENABLE_EXTENSION_AUTOLOADING= -DENABLE_EXTENSION_AUTOINSTALL= -DLOCAL_EXTENSION_REPO=""  -DOVERRIDE_GIT_DESCRIBE=""  -DCMAKE_BUILD_TYPE=Release ../.. && \
cmake --build . --config Release
-- git hash d58cc56f85, version v1.1.2-dev44, extension folder d58cc56f85
-- Extensions will be deployed to: /media/user/H3D2P2/Test/duckdb/duckdb/build/release/repository
-- Load extension 'parquet' from '/media/user/H3D2P2/Test/duckdb/duckdb/extensions' @ d58cc56f85
-- Extensions linked into DuckDB: [parquet]
-- Configuring done
-- Generating done
-- Build files have been written to: /media/user/H3D2P2/Test/duckdb/duckdb/build/release
gmake[1]: Entering directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
gmake[2]: Entering directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
gmake[3]: Entering directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
gmake[3]: Leaving directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
[  0%] Built target duckdb_platform_binary
gmake[3]: Entering directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
gmake[3]: Leaving directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
gmake[3]: Entering directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
/bin/sh: 1: duckdb_platform_binary: not found
Provide explicit DUCKDB_PLATFORM=your_target_arch to avoid build-type detection of the platform
gmake[3]: *** [CMakeFiles/duckdb_platform.dir/build.make:70: CMakeFiles/duckdb_platform] Error 1
gmake[3]: Leaving directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
gmake[2]: *** [CMakeFiles/Makefile2:3647: CMakeFiles/duckdb_platform.dir/all] Error 2
gmake[2]: Leaving directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
gmake[1]: *** [Makefile:136: all] Error 2
gmake[1]: Leaving directory '/media/user/H3D2P2/Test/duckdb/duckdb/build/release'
make: *** [Makefile:302: release] Error 2

**Comment by Tishj** [comment]
I believe PRs are always welcome to add new builds to the CI release pipeline
